src/index.js

```diff
@@ -230,4 +225,51 @@ function strWithLen(len) {
   return new Array(len + 1).join( 'x' );
 }
 
-export { rules };
+function getTagName({ env, tagNameOption}) {
+  if (tagNameOption) {
+    return tagNameOption;
+  } else if (env === 'relay') {
+    return 'Relay.QL';
+  }
+
+  return 'gql';
+}
+
+function getConfig(context) {
+  const {
+    schemaJson, // Schema via JSON object
+    schemaJsonFilepath, // Or Schema via absolute filepath
+    env,
+    tagName: tagNameOption,
+  } = context.options[0];
+
+  return {
+    schemaJson,
+    schemaJsonFilepath,
+    env,
+    tagNameOption
+  }
+}
+
+const gqlProcessor = {
+  preprocess: function(text, filename) {
+    const { env, tagNameOption } = getConfig(globalContext);
+    const tagname = getTagName({ env, tagNameOption });
+
+    return [`${tagname}\`${text}\``];
```

Comment:
Should backticks be escaped from the GraphQL file contents? Otherwise a backtick in a e.g. a GraphQL comment or string might cause the preprocessed template literal to be terminated early.